Add specs for controller requests and helpers

otherdcn opened the pull request:
As requested, I added controller request and helper specs.

What do you think?

otherdcn committed:
Add specs for controller requests and helpers

javiros commented on spec/helpers/application_helper_spec.rb:
```diff
@@ -0,0 +1,20 @@
+      api_key = Rails.application.credentials.api_sports.football[:key]
+      api_football_status_url = "https://v3.football.api-sports.io/status"
+
+      ApplicationHelper::APIFootball.get(api_football_status_url, headers: { "x-apisports-key" => api_key })
```
we should probably use a different approach here
1. use VCR cassettes to record the API response so we don't need to make external calls every time we run the specs, it is expensive to do so
2. We should mock the API response and stub the call to the API

we can have a call to work on this if you want

javiros commented on spec/helpers/countries_helper_spec.rb:
```diff
@@ -0,0 +1,39 @@
+
+RSpec.describe CountriesHelper, type: :helper do
+  describe "#countries" do
+    def get_countries(country_code: nil)
```
same here, we should not make an actual call to the service but mock its response instead

javiros commented on spec/helpers/leagues_helper_spec.rb:
```diff
@@ -0,0 +1,48 @@
+
+RSpec.describe LeaguesHelper, type: :helper do
+  describe "#countries" do
+    def get_leagues(country_code: nil)
```
not sure defining methods in specs is a sound approach. we should review this on a 1:1 video call

otherdcn replied:
Yes, let's do a video call. It would really help me.

Whenever you're free.

javiros reviewed (changes requested):
Lets work together on these specs, a change of approach is required 👍🏼

javiros committed:
Changing CountriesHelper

javiros committed:
Refactor spec

javiros committed:
More spec refactoring

otherdcn committed:
Add VCR and Webmock gems for mock testing

Configure and use VCR + Webmock for API reuqests and testing API helper
methods.
